core/src/test/java/io/questdb/test/tools/TestUtils.java

```diff
@@ -811,6 +811,68 @@ public static void assertResponse(HttpClient.Request request, int expectedStatus
         }
     }
 
+    /**
+     * Asserts that the actual lines are the same as the expected lines in the reverse order.
+     * For example, assertion with expected: "123\n456\n789\n" and actual: "789\n456\n123\n" passes.
+     * This method expects for the char sequences to end up with a newline character.
+     */
+    public static void assertReverseLinesEqual(@Nullable String message, CharSequence expected, CharSequence actual) {
+        if (expected == null && actual == null) {
+            return;
+        }
+
+        if (expected != null && actual == null) {
+            Assert.fail("Expected: \n`" + expected + "`but have NULL");
+        }
+
+        if (expected == null) {
+            Assert.fail("Expected: NULL but have \n`" + actual + "`\n");
+        }
+
+        if (expected.length() != actual.length()) {
+            Assert.assertEquals(message, reverseLines(expected), actual);
+        }
+
+        if (expected.length() == 0) {
+            return;
+        }
+
+        assert expected.charAt(expected.length() - 1) == '\n' : "Expected lines must end with a newline character.";
+        assert actual.charAt(actual.length() - 1) == '\n' : "Actual lines must end with a newline character.";
+
+        int expLo = expected.length();
+        int actHi = -1;
+        final int actLen = actual.length();
+        while (expLo != 0) {
+            int idx = Chars.lastIndexOf(expected, 0, expLo - 1, "\n");
+            int len;
+            if (idx == -1) {
+                len = expLo;
+                expLo = 0;
+            } else {
+                len = expLo - idx - 1;
+                expLo = idx + 1;
+            }
+
+            final int actLo = actHi + 1;
+            idx = Chars.indexOf(actual, actLo, actLen, "\n");
+            if (idx == -1 || idx - actHi != len) {
+                Assert.assertEquals(message, reverseLines(expected), actual);
+            }
+            actHi = idx;
+
+            for (int j = 0; j < len; j++) {
+                if (expected.charAt(expLo + j) != actual.charAt(actLo + j)) {
+                    Assert.assertEquals(message, reverseLines(expected), actual);
```

Comment:
this is strange, if we here, isn't it already known expected and actual are different?